src/Digdir.Domain.Dialogporten.Application/Features/V1/ResourceRegistry/Commands/Synchronize/SynchronizeResourceRegistryCommand.cs

```diff
@@ -0,0 +1,142 @@
+        IResourceRegistry resourceRegistry,
+        ILogger<SynchronizeResourceRegistryCommandHandler> logger)
+    {
+        _db = db ?? throw new ArgumentNullException(nameof(db));
+        _unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
+        _resourceRegistry = resourceRegistry ?? throw new ArgumentNullException(nameof(resourceRegistry));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+    public async Task<SynchronizeResourceRegistryResult> Handle(SynchronizeResourceRegistryCommand request, CancellationToken cancellationToken)
+    {
+        // 1. Get the last updated timestamp from parameter, or the database, or use a default
+        var lastUpdated = request.Since
+            ?? (await _db.SubjectResourceLastUpdates
+                .OrderByDescending(x => x.LastUpdate)
+                .FirstOrDefaultAsync(cancellationToken))
+                ?.LastUpdate
+            ?? DateTimeOffset.MinValue;
+
+        // Allow for a few seconds of drift between server clocks
+        var newLastUpdated = DateTimeOffset.Now.Subtract(TimeSpan.FromSeconds(3));
+
+        _logger.LogInformation("Fetching updated subject resources since {LastUpdated}", lastUpdated);
+
+        // 2. Get the resources from the resource registry, supplying the last updated as "since"
+        var remoteUpdatedSubjectResources =
+            await _resourceRegistry.GetUpdatedSubjectResources(lastUpdated, cancellationToken);
+
+        _logger.LogInformation("Fetched {Count} updated subject resources", remoteUpdatedSubjectResources.Count);
+
+        // 3. Delete the subjectResources that are no longer in the resource registry
+        var remoteDeletedSubjectResources = remoteUpdatedSubjectResources
+            .Where(x => x.Deleted)
+            .Select(deletedSubjectResource => new
+            {
+                deletedSubjectResource.Resource,
+                deletedSubjectResource.Subject
+            })
+            .ToList();
+
+        if (remoteDeletedSubjectResources.Count != 0)
+        {
+            var resourceStrings = remoteDeletedSubjectResources.Select(r => r.Resource.ToString()).ToList();
+            var subjectStrings = remoteDeletedSubjectResources.Select(r => r.Subject.ToString()).ToList();
+
+            var localSubjectResourcesToDelete = await _db.SubjectResources
+                .Where(x => resourceStrings.Contains(x.Resource) && subjectStrings.Contains(x.Subject))
+                .ToListAsync(cancellationToken);
+
+            if (localSubjectResourcesToDelete.Count != 0)
+            {
+                _logger.LogInformation("Deleting {Count} subject resources", localSubjectResourcesToDelete.Count);
+                _db.SubjectResources.RemoveRange(localSubjectResourcesToDelete);
+            }
+        }
+
+        // 4. Add the ones that are new (or no longer deleted)
+        var remoteNewSubjectResources = remoteUpdatedSubjectResources
```

Comment:
Ja, men det kan teoretisk være flere UpdateSubjectResource som representerer den samme SubjectResource'n i _denne_ updaten. Da er det den nyeste som vinner. Tror vi må gruppere de vi får fra ressurs registeret på ressurs og subjekt og dra ut den siste fra hver gruppe for å vite hva som faktisk er nyeste. Om jeg har forstått dette korrekt.

Reply:
Hvert par er unikt.